### PR TITLE
Add episode length for Bit Flipping Env

### PR DESCRIPTION
**Pull request by sriram13m**
Add episode length for Bit Flipping Environment

### CODE REVIEW

**Comment by findmyway** [comment]
If I understand it correctly, the length is used to force terminating the game.
If so, I think one can use the [MaxTimeoutWrapper](https://github.com/JuliaReinforcementLearning/ReinforcementLearningEnvironments.jl/blob/master/src/environments/wrappers/MaxTimeoutEnv.jl) contributed by @Sid-Bhatia-0

**Comment by sriram13m** [comment]
> If I understand it correctly, the length is used to force terminating the game.

The length(max_steps) is implicitly assumed to be the number of bits unless otherwise stated. ( `The length of each episode was equal to the number of bits and the episode was considered successful if the goal state was achieved at an arbitrary time step during the episode` from [HER paper](https://arxiv.org/pdf/1707.01495.pdf))

> If so, I think one can use the [MaxTimeoutWrapper](https://github.com/JuliaReinforcementLearning/ReinforcementLearningEnvironments.jl/blob/master/src/environments/wrappers/MaxTimeoutEnv.jl) contributed by @Sid-Bhatia-0

 I think using MaxTimeoutWrapper forgo the assumption and one needs to explicitly state `length`

**Comment by findmyway** [comment]
I see. Thanks for your explanation.

File: src/environments/examples/BitFlippingEnv.jl [hunk 1]
```diff
@@ -6,17 +6,43 @@ In Bit Flipping Environment we have n bits. The actions are 1 to n where executi
...
+    max_steps = N
+    BitFlippingEnv(N, rng, state, goal_state, max_steps, 0)
+end
+
+function BitFlippingEnv(; N = 8, T = 8)
+    rng = Random.GLOBAL_RNG
+    state = bitrand(rng, N)
+    goal_state = bitrand(rng, N)
+    max_steps = T
+    BitFlippingEnv(N, rng, state, goal_state, max_steps, 0)
```
Comment (findmyway) [inline src/environments/examples/BitFlippingEnv.jl]: I think these two methods will be overridden by the next constructor (In Julia, methods are only dispatched by positional arguments).

We can only define one constructor:

```julia
function BitFlippingEnv(; N = 8, T=N, rng = Random.GLOBAL_RNG)
...
end
```

**Review by findmyway — CHANGES_REQUESTED** [review]

**Pull request merged**